src/v8/v8.cc

```diff
@@ -263,9 +268,58 @@ bool V8::load(const std::string &code, bool allow_precompiled) {
     assert((shared_module_ != nullptr));
   }
 
+  buildFunctionNameIndex();
+
   return module_ != nullptr;
 }
 
+void V8::buildFunctionNameIndex() {
+  // build function index -> function name map for backtrace
+  // https://webassembly.github.io/spec/core/appendix/custom.html#binary-namesubsection
+  auto name_section = getCustomSection("name");
+  if (name_section.size()) {
+    const byte_t *pos = name_section.data();
+    const byte_t *end = name_section.data() + name_section.size();
+    while (pos < end) {
+      if (*pos++ != 1) {
+        pos += parseVarint(pos, end);
+      } else {
+        const auto size = parseVarint(pos, end);
+        if (size == static_cast<uint32_t>(-1) || pos + size > end) {
+          function_names_index_ = {};
+          return;
+        }
+        const auto start = pos;
+        const auto namemap_vector_size = parseVarint(pos, end);
+        if (namemap_vector_size == static_cast<uint32_t>(-1) || pos + namemap_vector_size > end) {
+          function_names_index_ = {};
+          return;
+        }
+        for (auto i = 0; i < namemap_vector_size; i++) {
+          const auto func_index = parseVarint(pos, end);
+          if (func_index == static_cast<uint32_t>(-1) || pos + func_index > end) {
```

Comment:
This isn't checking size, so `pos + func_index > end` is wrong here.

Reply:
yeah